### DISCUSSION TITLE
Handling multiple trained pipeline packages

### DISCUSSION THREAD

**Discussion by paulbriton** [explosion/spaCy#7377] [Help: Best practices]
Hello!
I'm looking for the most pythonic/recommended way to handle multiple languages with spaCy on a fastAPI application. I would like to send a request to the API containing the ISO code of the language and conditionally load the corresponding model. 

So far here are the two solutions I've found:
```python
LANGUAGE_MODEL = {
  "en": "en_core_web_sm",
  "pt": "pt_core_news_sm",
  # ...
}

import spacy
nlp = spacy.load(LANGUAGE_MODEL[lang])
``` 
```python
import en_core_web_sm
import pt_core_news_sm
# ...

if lang == "en":
  nlp = en_core_web_sm.load()
elif lang == "pt":
  nlp = en_core_web_sm.load()
```

Is there a better one? What do you recommend?

**Comment by polm** [comment]
If you actually want to conditionally load the model, I would recommend the first pattern because it's structurally simpler. The second one requires you to type the same information repeatedly, and can more readily cause typos, like the one in your example.

That said, loading a model per request is probably going to be really slow. What I would do for a multi-lingual web application, assuming you can fit multiple models in memory, is load the models and keep them around to re-use across requests. Something like this:

```
import spacy
nlp = {}
nlp["en"] = spacy.load("en_core_news_sm")
nlp["pt"] = spacy.load("pt_core_news_sm")

doc = nlp[lang](text)
```

**Reply by paulbriton** [reply]
Thanks for the feedback! You're totally right about not loading the model at each request.

My only concern is about the memory usage of loading about 15 models into memory, so I might end up loading some of them conditionally anyway.
